packages/jbrowse-desktop/public/electron.js

```diff
@@ -0,0 +1,239 @@
+try {
+  fs.statSync(sessionDirectory)
+} catch (error) {
+  if (error.code === 'ENOENT' || error.code === 'ENOTDIR')
+    fs.mkdirSync(sessionDirectory, { recursive: true })
+  else throw error
+}
+
+let mainWindow
+
+function createWindow() {
+  mainWindow = new BrowserWindow({
+    width: 1400,
+    height: 800,
+    webPreferences: {
+      preload: isDev
+        ? path.join(app.getAppPath(), 'public/preload.js')
+        : `file://${path.join(app.getAppPath(), 'public/../build/preload.js')}`,
+    },
+  })
+  mainWindow.loadURL(
+    isDev
+      ? url.format(devServerUrl)
+      : `file://${path.join(app.getAppPath(), 'public/../build/index.html')}`,
+  )
+  mainWindow.setMenuBarVisibility(false)
+  if (isDev) {
+    // Open the DevTools.
+    // BrowserWindow.addDevToolsExtension('<location to your react chrome extension>');
+    // mainWindow.webContents.openDevTools()
+  }
+  mainWindow.on('closed', () => {
+    BrowserWindow.getAllWindows().forEach(win => win.close())
+    mainWindow = null
+  })
+}
+
+app.on('ready', createWindow)
+
+app.on('window-all-closed', () => {
+  if (process.platform !== 'darwin') {
+    app.quit()
+  }
+})
+
+app.on('activate', () => {
+  if (mainWindow === null) {
+    createWindow()
+  }
+})
+
+ipcMain.on('createWindowWorker', event => {
+  const workerWindow = new BrowserWindow({
+    show: false,
+    webPreferences: {
+      preload: isDev
+        ? path.join(app.getAppPath(), 'public/preload.js')
+        : `file://${path.join(app.getAppPath(), 'public/../build/preload.js')}`,
+    },
+  })
+  workerWindow.loadURL(
+    isDev
+      ? url.format({ ...devServerUrl, pathname: 'worker.html' })
+      : `file://${path.join(app.getAppPath(), 'public/../build/worker.html')}`,
+  )
+  // workerWindow.webContents.openDevTools()
+  event.returnValue = workerWindow.id
+})
+
+ipcMain.answerRenderer('loadConfig', async () => {
+  let configJSON
+  try {
+    configJSON = await fsReadFile(configLocation, { encoding: 'utf8' })
+  } catch (error) {
+    if (error.code === 'ENOENT') {
+      // make a config file since one does not exist yet
+      const configTemplateLocation = isDev
+        ? path.join(app.getAppPath(), 'public/test_data/config.json')
+        : `file://${path.join(
+            app.getAppPath(),
+            'public/../build/test_data/config.json',
+          )}`
+      await fsCopyFile(configTemplateLocation, configLocation)
+      configJSON = await fsReadFile(configLocation, { encoding: 'utf8' })
+    } else throw error
+  }
+  return JSON.parse(configJSON)
+})
+
+ipcMain.answerRenderer('saveConfig', async configSnapshot => {
+  return fsWriteFile(configLocation, JSON.stringify(configSnapshot))
+})
+
+ipcMain.answerRenderer('listSessions', async () => {
+  const sessionFiles = await fsReaddir(sessionDirectory)
+  const sessionFilesData = []
+  for (const sessionFile of sessionFiles) {
+    if (sessionFile.endsWith('.thumbnail'))
+      sessionFilesData.push(
```

Comment:
For the .thumbnail files we want the actual file (to display the image in the start screen), but for the session JSON files we don't actually need the file yet, just the stats, so it does and fs.stat instead of an fs.readFile